pkg/devfile/parser/data/v2/attributes.go

```diff
@@ -17,32 +17,36 @@ func (d *DevfileV2) GetAttributes() (attributes.Attributes, error) {
 	}
 }
 
-// UpdateAttributes updates the devfile top level attributes
-func (d *DevfileV2) UpdateAttributes(attr attributes.Attributes) error {
+// UpdateAttributes updates the devfile top level attribute for the specific key, err out if key is absent
+func (d *DevfileV2) UpdateAttributes(key string, value interface{}) error {
+	var err error
+
 	// This feature was introduced in 2.1.0; so any version 2.1.0 and up should use the 2.1.0 implementation
 	switch d.SchemaVersion {
 	case "2.0.0":
 		return fmt.Errorf("top-level attributes is not supported in devfile schema version 2.0.0")
 	default:
-		d.Attributes = attr
+		if d.Attributes.Exists(key) {
+			d.Attributes.Put(key, value, &err)
+		} else {
+			return fmt.Errorf("cannot update top-level attribute, key %s is not present", key)
+		}
 	}
 
-	return nil
+	return err
```

Comment:
no need for err. just return nil here or in the if statement. Same for the add function

Reply:
we are using it here, if i did return nil there would be one more `if err != nil` check which is the last execution in that if block. So return err is the same thing, pls check here https://github.com/devfile/library/pull/82/commits/28cfac00e0399b7fdf10700a3e9a26e3bd6d01c6#diff-ab03248fb97b24c81ae668b35d88bb6f87073ee6739cf4b6dd754d86fb685b32R30